nightingale_navigation/hallway_planner/src/hallway_planner.cpp

```diff
@@ -0,0 +1,8 @@
+/*********************************************************************
+ * Software License Agreement (MIT License)
+ * Author: Younes Reda
+ *********************************************************************/
+#include <pluginlib/class_list_macros.hpp>
+#include <hallway_planner/hallway_planner.hpp>
+
+PLUGINLIB_EXPORT_CLASS(hallway_planner::HallwayPlanner, nav_core::BaseGlobalPlanner)
```

Comment:
could you send the documentation on how the plugin stuff works? I don't think it's worth you explaining it to me so I can read up on it

Reply:
i kinda just copied the globalplanner code and defined my own plugin but this link says more http://wiki.ros.org/pluginlib